lua/autorun/sh_waiter.lua

```diff
@@ -9,18 +9,20 @@ Waiter.lastLoopDuration = 0
 local patronQueue = {}
 local patronCount = 0
 
-local maxAttempts = 10
+local defaultMaxAttempts = 10
 
--- Minimum delay between executions of attendPatrons()
-local minBreakAfterAttending = 1
+-- Minimum delay (in seconds) between executions of attendPatrons()
+local minBreakAfterAttending = 0.5
 
-local function generatePatron( waitingFor, onSuccess, onTimeout )
+local function generatePatron( waitingFor, onSuccess, onTimeout, maxAttempts )
     print( "[CFC Waiter] Generating new Patron!" )
+
     local patron = {}
-    patron["onSuccess"] = onSuccess
-    patron["onTimeout"] = onTimeout
-    patron["waitingFor"] = waitingFor
-    patron["attempts"] = 0
+    patron.onSuccess = onSuccess
+    patron.onTimeout = onTimeout
+    patron.waitingFor = waitingFor
+    patron.maxAttempts = maxAttempts or defaultMaxAttempts
+    patron.attempts = 0
```

Comment:
And these lines would be something like:
```lua
patron.onSuccess = patronArg.onSuccess
patron.onTimeout = patronArg.onTimeout
patron.waitingFor = patronArg.waitingFor
patron.maxAttempts = patronArg.maxAttempts or defaultMaxAttempts
patron.attempts = 0
```

Or even 

```lua
local patron = patronArg.copy()
patron.maxAttempts = patronArg.maxAttempts or defaultMaxAttempts
```

I'm in class on mobile, pls forgive